components/base/kinematicbase/fake_kinematics.go

```diff
@@ -206,19 +299,27 @@ func (fk *fakePTGKinematics) GoToInputs(ctx context.Context, inputs []referencef
 	fk.lock.Lock()
 	fk.origin = referenceframe.NewPoseInFrame(fk.origin.Parent(), spatialmath.Compose(fk.origin.Pose(), newPose))
 	fk.lock.Unlock()
-	time.Sleep(time.Duration(fk.sleepTime) * time.Millisecond)
-	return nil
+	timeoutCtx, cancelFn := context.WithTimeout(ctx, time.Millisecond*time.Duration(fk.sleepTime))
+	defer cancelFn()
+	<-timeoutCtx.Done()
```

Comment:
This approach makes it so that we return early if the context is cancelled before the sleep timeout, thus preventing it from hanging when the context is cancelled.